web/modules/custom/lit_cover_service/src/Command/LitCoverServiceCommands.php

```diff
@@ -0,0 +1,282 @@
+   *
+   * @command lit_cover_service:add_or_replace_covers
+   * @aliases add_or_replace_covers
+   */
+  public function addOrReplaceCovers($limit = self::LIMIT_ALL) {
+    $limit = $this->parseLimit($limit);
+
+    $this->loggerChannelFactory->get('lit_cover_service')->info('Fetch covers batch operations start');
+
+    $nids = $this->getBookNids($limit);
+
+    if (!empty($nids)) {
+      $chunks = array_chunk($nids, self::BATCH_SIZE);
+
+      $numOperations = count($nids);
+      $batchId = 1;
+      $batchTotal = count($chunks);
+
+      $operations = [];
+      foreach ($chunks as $batchNids) {
+        $operations[] = [
+          '\Drupal\lit_cover_service\Service\BatchService::replaceBookCovers',
+          [
+            $batchId,
+            $batchTotal,
+            $batchNids
+          ],
+        ];
+        ++$batchId;
+      }
+
+      $batch = [
+        'title' => t('Fetching covers for @num "book" node(s)', ['@num' => $numOperations]),
+        'operations' => $operations,
+        'finished' => '\Drupal\lit_cover_service\Service\BatchService::deleteBookCoversFinished',
+      ];
+      batch_set($batch);
+      drush_backend_batch_process();
+      $this->loggerChannelFactory->get('lit_cover_service')->info('Fetch or replace covers batch operations end.');
+      $this->logger()->success("Book covers fetched");
+    }
+    else {
+      $this->logger()->warning('No nodes of this type @type', ['@type' => 'book']);
+    }
+  }
+
+  /**
+   * Get node ids for all 'Book' nodes.
+   *
+   * @param int $limit
+   *
+   * @param string $cover
+   *
+   * @return array
+   */
+  private function getBookNids(int $limit, string $cover = 'all'): array
+  {
+    try {
+      $storage = $this->entityTypeManager->getStorage('node');
+      $query = $storage->getQuery()
+        ->condition('type', 'book')
+        ->sort('nid', 'DESC');
+
+      switch ($cover) {
+        case 'all':
+          break;
+        case self::HAS_COVER:
+          $query->exists('field_book_cover_image');
+          break;
+        case self::NO_COVER:
+          $query->notExists('field_book_cover_image');
+          break;
+        default:
+          throw new \InvalidArgumentException('Unknown value' . $cover);
+      }
+
+      if ($limit) {
+        $query->range(0, $limit);
+      }
+      return $query->execute();
+    }
+    catch (\Exception $e) {
+      $this->logger->error('Error found @e', ['@e' => $e]);
+    }
+  }
+
+  /**
+   * Parse string argument to integer value.
+   *
+   * @param string $limit
+   *
+   * @return int
+   */
+  private function parseLimit(string $limit): int
+  {
+    if (self::LIMIT_ALL === $limit) {
+      $intLimit = 0;
+    } else if (is_numeric($limit)) {
+      $intLimit = intval($limit);
+      // Returns the integer value of var on success, or 0 on failure.
+      if (0 === $intLimit) {
+        throw new \InvalidArgumentException($limit . ' must be an integer value');
+      }
+    } else {
```

Comment:
Drupal coding style else / else if should start on a new line.

Reply:
Ugly as hell but we gotta standard

Reply:
I think it is way better that this; on the same line.... :-)